rad/src/filters/mode.rs

```diff
@@ -0,0 +1,208 @@
+use crate::{
+    error::RadError,
+    reducers::mode::mode,
+    types::{array::RadonArray, RadonType, RadonTypes},
+};
+use witnet_data_structures::radon_report::{ReportContext, Stage};
+
+pub fn mode_filter(
+    input: &RadonArray,
+    context: &mut ReportContext,
+) -> Result<RadonTypes, RadError> {
+    let mode = mode(&input)?;
+    let mut liars = vec![];
+
+    let filtered_vec: Vec<RadonTypes> = input
+        .value()
+        .into_iter()
+        .filter(|rad_types| {
+            let cond = rad_types == &mode;
+            liars.push(!cond);
+            cond
+        })
+        .collect();
+
+    if let Stage::Tally(ref mut metadata) = context.stage {
+        metadata.update_liars(liars);
+    }
+
+    Ok(RadonArray::from(filtered_vec).into())
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use crate::types::{integer::RadonInteger, string::RadonString};
+    use witnet_data_structures::radon_report::TallyMetaData;
+
+    // Helper function which works with Rust integers, to remove RadonTypes from tests
+    fn imode(input_i128: &[i128], ctx: &mut ReportContext) -> Result<Vec<i128>, RadError> {
+        let input_vec: Vec<RadonTypes> = input_i128
+            .iter()
+            .map(|f| RadonTypes::Integer(RadonInteger::from(*f)))
+            .collect();
+        let input = RadonArray::from(input_vec);
+
+        let output = mode_filter(&input, ctx)?;
+
+        let output_vec = match output {
+            RadonTypes::Array(x) => x.value(),
+            _ => panic!("Filter method should return a RadonArray"),
+        };
+        let output_i128 = output_vec
+            .into_iter()
+            .map(|r| match r {
+                RadonTypes::Integer(x) => x.value(),
+                _ => panic!("Filter method should return an array of integers"),
+            })
+            .collect();
+
+        Ok(output_i128)
+    }
+
+    #[test]
+    fn test_filter_mode_empty() {
+        let input = vec![];
+        let expected = RadError::ModeEmpty;
```

Comment:
A filter applied to an empty array could just return an empty array, and the error would naturally appear when trying to reduce an empty array.

Reply:
The error is gonna appear anyway, so I think is better to handle as soon as possible